CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+
+In order to foster an open, inclusive, and welcoming community, all contributors agree to adhere to [the Astropy code of conduct](https://www.astropy.org/code_of_conduct.html), which we adopt in this project.
+
+
+## Contributing code using Pull Requests
+
+Code contributions are most welcome. You can in particular look for GitHub issues marked as `contributions welcome` or `good first issue`. But you can also propose adding a new functionality, in which case you may find it beneficial to first open a GitHub issue to discuss the feature you want to implement ahead of opening a Pull Request.
+
+
+Once you have some code you wish to contribute, you can follow this procedure to prepare a Pull Request:
+
+- Fork the JAX-GalSim repository under your own account, using the **Fork** button on the top right of the [GitHub page](https://github.com/GalSim-developers/JAX-GalSim).
+
+- Clone and pip install your fork of the repository like so:
+  ```bash
+  git clone https://github.com/YOUR_USERNAME/JAX-GalSim
+  cd JAX-GalSIM
+  pip install --user -e .
+  ```
+  This will install your local fork in editable mode, meaning you can directly modify source files in this folder without having to reinstall the package for them to be taken into account.
+
+- Open a branch for your developments:
+  ```bash
+  git checkout -b name-that-describes-my-feature
+  ```
+
+- Add your changes to the code using your favorite editor. You may at any moment test that everything is still working by running the test suite. From the root folder of the repository, run:
+  ```bash
+  pytest
+  ```
+
+- Once you are happy with your modifications, commit them, and push your changes to GitHub:
+  ```python
+  git add file_I_changed.py
+  git commit -m "a message that describes your modifications"
+  git push -set-upstream origin name-that-describes-my-feature
+  ```
+
+- From your GitHub interface, you should now be able to open a Pull Request to the JAX-GalSim repository. 
+
+Before submitting your PR, have a look at the procedure documented below.
+
+### Checklist before opening a Pull Request
+
+- Pull Requests should be self-contained and limited in scope, otherwise they become too difficult to review. If your modifications are broad, consider opening several smaller Pull Requests.
+
+- Make sure your fork and branch are up-to-date with the `main` branch of JAX-GalSim. To update your local branch, you may do so from the GitHub interface, or you may use this CLI command:
+  ```bash
+  # Only needs to be done once:
+  git remote add upstream http://www.github.com/GalSim-developers/JAX-GalSim
+  # This will update your local branch
+  git fetch upstream
+  git rebase upstream/main
+  ```
+
+- Make sure the unit tests still work:
+  ```bash
+  pytest
+  ```
+  Ideally there should be some new unit tests for the new functionality, unless the work is completely covered by existing unit tests.
+
+- Make sure your code conforms to the [Black](https://github.com/psf/black) style:
```

Comment:
Not a fan.  But you do you.

Reply:
:-) I don't have super strong feelings about this, I just want to have automated formatting. If there are other formats/tools you prefer let me know. In other projects I also `yapf` which has a lot of flexibility to define a style sheet.